madanalysis/layout/plotflow_for_dataset.py

```diff
@@ -84,58 +88,133 @@ def CreateHistogram(self):
 
+                # integral
+                integral = (
+                    self.histos[iplot].positive.integral
+                    - self.histos[iplot].negative.integral
+                )
+                multiweight_integral, multiweight_eff = 0, 0
+                if len(self.multiweight_histos) != 0:
+                    multiweight_integral = self.multiweight_histos[iplot].integral
+
+                # compute efficiency : Nevent / Ntotal
+                if self.dataset.measured_global.nevents == 0:
+                    eff = 0
+                else:
+                    nevt = float(self.dataset.measured_global.nevents)
+                    eff = (
+                        self.histos[iplot].positive.nevents
+                        + self.histos[iplot].negative.nevents
+                    ) / nevt
+                    if self.multiweight_histos[iplot]:
+                        multiweight_eff = self.multiweight_histos[iplot].nevents / nevt
+
                 # compute the good xsection value
                 thexsection = self.xsection
                 if self.main.normalize == NormalizeType.LUMI_WEIGHT:
                     thexsection = thexsection * self.dataset.weight
 
-                scale = processor.scale(lumi=self.main.lumi)
+                # compute final entries/event ratio
+                entries_per_events = 0
+                sumw = self.histos[iplot].positive.sumw - self.histos[iplot].negative.sumw
+                Nentries = (
+                    self.histos[iplot].positive.sumwentries
+                    - self.histos[iplot].negative.sumwentries
+                )
+                if sumw != 0 and Nentries != 0:
+                    entries_per_events = sumw / Nentries
+
+                # compute the scale
+                if integral != 0:
+                    scale = (
+                        thexsection
+                        * self.main.lumi
+                        * 1000
+                        * eff
+                        * entries_per_events
+                        / integral
+                    )
+                else:
+                    scale = 1  # no scale for empty plot
+
+                if self.multiweight_histos[iplot]:
+                    entries_per_events = 0
+                    sumw = self.multiweight_histos[iplot].central_sumw_over_events
+                    Nentries = self.multiweight_histos[iplot].central_sumw_over_entries
+                    if Nentries != 0:
+                        entries_per_events = sumw / Nentries
+                    # compute the scale
+                    if multiweight_integral != 0:
+                        multiweight_scale = (
+                            thexsection
+                            * self.main.lumi
+                            * 1000
+                            * multiweight_eff
+                            * entries_per_events
+                            / multiweight_integral
```

Comment:
Same comment as above. I think that we need to use "integral" instead of "multiweight_integral".

Reply:
But integral is not same as multiweight integral. This integral is based on 0th weight so its not necessarily the nominal weight. The multiweight integral computes the integral of the nominal weight.

Reply:
The integral must be based on what we define as the central weight.

Reply:
for both definitions it is, but they are different from each other. we have two histograms here old version and new version. Old version takes the 0th weight new one takes the nominal one

Reply:
Ah yes, this is what I forgot. Then it is fine as the old one will disappear at some point.